rclcpp/test/rclcpp/test_utilities.cpp

```diff
@@ -78,3 +83,130 @@ TEST(TestUtilities, multi_init) {
   EXPECT_FALSE(rclcpp::ok(context1));
   EXPECT_FALSE(rclcpp::ok(context2));
 }
+
+TEST(TestUtilities, test_context_basic_access) {
+  auto context1 = std::make_shared<rclcpp::contexts::DefaultContext>();
+  EXPECT_NE(nullptr, context1->get_init_options().get_rcl_init_options());
+  EXPECT_EQ(0u, context1->get_on_shutdown_callbacks().size());
+  EXPECT_EQ(std::string{""}, context1->shutdown_reason());
+}
+
+TEST(TestUtilities, test_context_basic_access_const_methods) {
+  auto context1 = std::make_shared<const rclcpp::contexts::DefaultContext>();
+
+  EXPECT_NE(nullptr, context1->get_init_options().get_rcl_init_options());
+  EXPECT_EQ(0u, context1->get_on_shutdown_callbacks().size());
+  // EXPECT_EQ(std::string{""}, context1->shutdown_reason()); not available for const
```

Comment:
This seems like a method that should be const, I wonder if there is a better thread-safe way of setting the string in `rclcpp::Context` so it can be.

Reply:
I imagine it is not const because the shutdown_reason may change. But, in terms of this specific test, is odd that the method is not available for a const context. Not sure how to proceed here

Reply:
It looks like it's const because it has to lock the mutex. It might just be worth opening an issue and getting some input from others. I don't know whether it's worth investigating or if it's not an important use case.

Reply:
I opened #1323 to track the conversation. I didn't tag anyone as it might not be a priority at the moment, let me know if you think I should tag someone in particular.